### DISCUSSION TITLE
octoprint instance on another machine

### DISCUSSION THREAD

**Discussion by mfuller59** [paukstelis/octoprint_deploy#157] [Ideas]
I discovered usbip is part of the linux kernel. It allows a usb port on one machine to be used by another machine.
I installed it on a pi3 as server and on a pc running debian as a client.
I had configured octoprint on the pi3 but wanted to bring it to the pc so all instances were centralised..
octoprint_deploy stumbles when finding the serial port so I copied the entire '.xxx' directory from pi3 to pc and adjusted haproxy to use ports 5001 & 5002 for my 2 instances.
copied the systemctl.service file and changed the service files accordingly.
Used a browser (tab1) to connect to instance 1. auto detect of local usb port works (cr6se printer)
Used a browser (tab2) to connect to instance 2. auto detect of remote usb port works (lowrider 3 cnc)

useful howto link when installing usbip:
https://developer.ridgerun.com/wiki/index.php?title=How_to_setup_and_use_USB/IP

**Comment by paukstelis** [comment]
I don't quite understand why you want to do this. If you have your printer/CNC connected to the RPi, it is going to make much more sense to just install multiple OctoPrint instances on that than dealing with serial-ip-serial latency, or the much greater possibility of dropped packets.

**Comment by mfuller59** [comment]
Thanks for your reply. The aim was to have 1 central pc with octoprint
controlling a printer and the cnc controller. Your comment about dropped
packets is a concern. I will rethink the concept. The cnc pi is on the
gantry with a usb connection to the cnc controller. Octoprint deploy is on
a pc on the other side of my garage with the printer making usb over
ethernet a solution.
Regards
Malcolm
